packages/protocol/contracts/verifiers/RiscZeroVerifier.sol

```diff
@@ -76,18 +68,21 @@ contract RiscZeroVerifier is EssentialContract, IVerifier {
         );
 
         // journalDigest is the sha256 hash of the hashed public input
-        bytes32 journalDigest = sha256(bytes.concat(hash));
+        bytes32 journalDigest = sha256(bytes.concat(FIXED_JOURNAL_HEADER, hash));
 
+        emit ProofVerified(_ctx.metaHash, journalDigest, hash);
         // call risc0 verifier contract
-        (bool success,) = address(receiptVerifier).staticcall(
-            abi.encodeCall(IRiscZeroReceiptVerifier.verify, (seal, imageId, journalDigest))
-        );
-
-        if (!success) {
+        try IRiscZeroVerifier(risk0Grouth16Verifier()).verify(seal, imageId, journalDigest) {
```

Comment:
Please use low level calls, then check for `success`.

`Try-catch` not the safest way to catch reverts. For example exceptions from calls to non-contracts are currently not catched by `try-catch`, for example.

Reply:
Thanks for your detailed explanation! I will fix it.